### PR TITLE
Backport m_opmoderated.cpp (cmode +U) from 2.1

### PR DESCRIPTION
**Pull request by GLolol**
Tested working. I imported the changes from my [own repository](https://github.com/OVERdrive-IRC/inspircd-modules/commits/master) as a squash merge, which includes the following:
- OVERdrive-IRC/inspircd-modules@c624c09 Change extban from `U:` to `u:` to prevent conflicts with m_services_account
- OVERdrive-IRC/inspircd-modules@1527108 Quick fixes so that the module compiles on InspIRCd 2.0

### CODE REVIEW

File: 2.0/m_opmoderated.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+		Implementation eventlist[] = { I_OnUserPreMessage, I_OnUserPreNotice, I_On005Numeric };
+		ServerInstance->Modules->Attach(eventlist, this, sizeof(eventlist)/sizeof(Implementation));
+	}
+
+	Version GetVersion()
+	{
+		return Version("Implements opmoderated channel mode +U (non-voiced messages sent to ops) and extban 'u'", VF_OPTCOMMON|VF_VENDOR);
+	}
+
+	ModResult DoMsg(User *user, void *dest, int target_type, std::string &text, char status, CUList &exempt_list, bool privmsg)
```
Comment (SaberUK) [inline 2.0/m_opmoderated.cpp]: The `*` and `&` should be next to the type not the name.

**Comment by GLolol** [comment]
Any update on this?

**Comment by SaberUK** [comment]
Other than my comment about moving the *& next to the type it looks good to me.

**Comment by attilamolnar** [comment]
Is this well tested?

**Comment by GLolol** [comment]
It works fine on my live network, though a side effect is that services clients and bots not trained to handle `@#channel` messages won't recognize anything sent from non-ops. Messages from non-ops to a `+U` channel is automatically changed to `@#channel`.

**Comment by siniStar7boy** [comment]
I'm also running this on a **production network** and so far have not had any issues.  It seems to work _perfectly_!   Can this be merged into inspircd-extras?

**Comment by attilamolnar** [comment]
Thanks, merged!

**Pull request merged**